functions/src/aqi-calculation/cleaned-reading.ts

```diff
@@ -38,7 +39,7 @@ function averageReadings(readings: Array<Pm25BufferElement>): BasicReading {
  * @param status - the status of the pm25Buffer (exists, does not exist, in progress)
  * @param bufferIndex - the next index to write to in the buffer
  * @param buffer - the pm25Buffer with the last 12 hours of data
- * @returns a BasicReading array of length 12 with the average PM2.5 value for each of the last 12 hours. If an hour lacks enough readings, then the entry for that hour is `undefined`.
+ * @returns a BasicReading array of length 12 with the average PM2.5 value for each of the last 12 hours. If an hour lacks enough readings, then the entry for that hour is null
```

Comment:
Why do we prefer `null` to `undefined` here?

Reply:
Just to be more clear/intentional about what each array element contains. `undefined` refers to an object that hasn't been initialized while `null` means that an object is missing, so I think `null` is more accurate here. The beginning of [this blog post](https://dmitripavlutin.com/7-tips-to-handle-undefined-in-javascript/) sort of discusses the difference. What do you think?

Reply:
Hmm. I think null is the right return value, but then I wonder if it wouldn’t be more appropriate to rather than fill with null and replace with actual, to initialize with undefined and replace with either actual or null. The outcome is the same, it’s just it feels like initializing with null goes against the idea that null is intentional absence.

Reply:
Sure that makes sense to me, I'll make the change